Jenkinsfile

```diff
@@ -30,8 +31,18 @@ def doc_only_change() {
     return rc == 1
 }
 
-def skip_stage(String stage) {
-    return cachedCommitPragma(pragma: 'Skip-' + stage) == 'true'
+def skip_stage(String stage, String def_val) {
+    return cachedCommitPragma(pragma: 'Skip-' + stage,
+                              def_val: def_val) == 'true'
+}
+
+def skip_stage(String stage, boolean def_val = false) {
+    String value = 'false'
+    if (def_val) {
+        value = 'true'
+    }
+    return cachedCommitPragma(pragma: 'Skip-' + stage,
+                              def_val: value) == 'true'
```

Comment:
Why do we need both of these implementations?

Even if we do need both, the first should not be re-implemented in the second but rather the second should call the first, to DRY out the code a bit.

Reply:
> I could not find a way to DRY them out with out re-writing the cachedCommitPragma in trusted-pipeline-lib

`cachedCommitPragma()` is defined in `Jenkinsfile`: https://github.com/daos-stack/daos/blob/5b3d28bcf2005ddaecfd1b61c37ccc9820179d67/Jenkinsfile#L109-L120

> 
> The cachedCommitPragma wants a string for the passed parameter. I think it looks better where it is called if it could also take a boolean parameter.

Even without doing anything to `cachedCommitPragma` you could do:

```suggestion
def skip_stage(String stage, String def_val) {
    return cachedCommitPragma(pragma: 'Skip-' + stage,
                              def_val: def_val) == 'true'
}

def skip_stage(String stage, boolean def_val = false) {
    String value = 'false'
    if (def_val) {
        value = 'true'
    }
    skip_stage(stage, value)
```

just to avoid duplicating lines 35-36 at lines 44-45.  This is a common method of implementing multiple signatures by using a flexible, common base method and then just building the different signatures on top of it.

But I still wonder why we even need the one with the `String` default value.  Why can't we just have the `boolean` one and change all of the calls in this `Jenkinsfile` to `boolean` values?